### ISSUE TITLE
Data Source: raw:com.google.step_count.delta:fitbit:charge-hr:io.pkp.fbit-gfit already exists

### ISSUE THREAD

**Issue by ronilaukkarinen** [praveendath92/fitbit-googlefit#57]
Haven't used it a while and noticed my cron job stopped working due to expired API key and updated python version. Authed again for both Google and Fitbit, but not working.

```` python
➜  fitbit-googlefit (master) /usr/local/bin/python3 /Users/rolle/fitbit-googlefit/app.py

Traceback (most recent call last):
  File "/Users/rolle/fitbit-googlefit/remote.py", line 119, in CreateGoogleFitDataSource
    dataSourceId=self.convertor.GetDataSourceId(dataType)).execute()
  File "/usr/local/lib/python3.6/site-packages/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/googleapiclient/http.py", line 840, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 404 when requesting https://www.googleapis.com/fitness/v1/users/me/dataSources/raw%3Acom.google.step_count.delta%3A191719253163%3Afitbit%3Acharge-hr%3Aio.pkp.fbit-gfit?alt=json returned "DataSourceId not found: raw:com.google.step_count.delta:191719253163:fitbit:charge-hr:io.pkp.fbit-gfit">

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/rolle/fitbit-googlefit/app.py", line 108, in <module>
    main()
  File "/Users/rolle/fitbit-googlefit/app.py", line 59, in main
    remote.CreateGoogleFitDataSource(dataType)
  File "/Users/rolle/fitbit-googlefit/remote.py", line 126, in CreateGoogleFitDataSource
    body=self.convertor.GetDataSource(dataType)).execute()
  File "/usr/local/lib/python3.6/site-packages/oauth2client/util.py", line 137, in positional_wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/googleapiclient/http.py", line 840, in execute
    raise HttpError(resp, content, uri=self.uri)
googleapiclient.errors.HttpError: <HttpError 409 when requesting https://www.googleapis.com/fitness/v1/users/me/dataSources?alt=json returned "Data Source: raw:com.google.step_count.delta:fitbit:charge-hr:io.pkp.fbit-gfit already exists">
````

Can you help me to resolve this?

**Comment by amrhassan** [comment]
Same error happens to me

**Comment by Grogdunn** [comment]
in "convertors.py" delete line 350 and 354 (projectNumber variable).
The bug was:

1. verify if the DataSouceId exist (raw:com.google.step_count.delta:191719253163:fitbit:charge-hr:io.pkp.fbit-gfit)
1. ah no it doesn't exist. Well create it. With id "raw:com.google.step_count.delta:fitbit:charge-hr:io.pkp.fbit-gfit"

look the differences:
raw:com.google.step_count.delta:**191719253163**:fitbit:charge-hr:io.pkp.fbit-gfit
raw:com.google.step_count.delta:fitbit:charge-hr:io.pkp.fbit-gfit

The script uses a "generated" id for one purpose and a different one for the same purpose.

**Comment by ronilaukkarinen** [comment]
Thanks @Grogdunn, it works with your instructions.

**Comment by praveendath92** [comment]
Resolved in #58
